AK/Math.h

```diff
@@ -249,8 +249,20 @@ constexpr T acos(T value)
 {
     CONSTEXPR_STATE(acos, value);
 
-    // FIXME: I am naive
-    return Pi<T> + asin(value);
+    T negation = (value < 0) ? 1 : 0;
+    T absolute_value = abs(value);
+
+    T result = static_cast<T>(-0.0187293);
+    result *= absolute_value;
+    result += static_cast<T>(0.0742610);
+    result *= absolute_value;
+    result -= static_cast<T>(0.2121144);
```

Comment:
is there a more verbose way to calculate these magic numbers, like asin does?

Reply:
Well, see page 159 of this 1955 paper:
https://blasingame.engr.tamu.edu/z_zCourse_Archive/P620_18C/P620_zReference/PDF_Txt_Hst_Apr_Cmp_(1955).pdf

I'll try to see if these values correspond to any of the `asin` constants.

Reply:
I could not find a good source for the constants used in that 1955 reference, and `asin` is implemented as a limited Taylor series, so I decided to make use of the mirroring between `acos` and `asin` in `y = PI / 2`.